### PR TITLE
Create Minio Interface Charm Library

### PR DESCRIPTION
**Pull request by DomFleischmann**
Add Minio Interface Charm Library which uses
Serialized Data Interface Library.

### CODE REVIEW

File: lib/charms/minio/v0/minio_interface.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+"""
+Minio Interface
+
+This Interface requires the serialized-data-interface package
+pleas add it to your requirements.txt
```
Comment (knkski) [inline lib/charms/minio/v0/minio_interface.py:5]: `s/pleas /please /`

File: src/charm.py [hunk 2]
```diff
@@ -34,18 +35,23 @@ def gen_pass() -> str:
...
         self.framework.observe(self.on.config_changed, self.set_pod_spec)
         self.framework.observe(self.on.minio_relation_joined, self.send_info)
 
     def send_info(self, event):
-        secret_key = get_or_set("password", default=gen_pass)
-        event.relation.data[self.unit]["service"] = self.model.app.name
-        event.relation.data[self.unit]["port"] = self.model.config["port"]
-        event.relation.data[self.unit]["access-key"] = self.model.config["access-key"]
-        event.relation.data[self.unit]["secret-key"] = secret_key
+        secret_key = get_or_set("password", configured=None, default=gen_pass)
```
Comment (knkski) [inline src/charm.py:46]: Can this actually get updated to match the other usage of `get_or_set`?

```
secret_key = get_or_set(
    "password",
    configured=self.model.config["secret-key"],
    default=gen_pass,
)
```
I just forgot to update this one when adding that argument

File: config.yaml [hunk 3]
```diff
@@ -1,13 +1,13 @@
 options:
   port:
-    type: string
-    default: '9000'
+    type: int
```
Comment (knkski) [inline config.yaml:3]: If this is string, we'll have to convert it to an int in the pod spec. If it's an int, we'll have to convert it to string in the relation data.

File: config.yaml [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
     description: HTTP port
   access-key:
     type: string
-    default: minio
+    default: 'minio'
     description: Access key
   secret-key:
     type: string
-    default: ~
+    default: ''
```
Comment (knkski) [inline config.yaml:12]: This can now be left as `~`

File: config.yaml (hunk 3 above)
Comment (DomFleischmann) [reply config.yaml:3]: That's the good part about using the serialized data interface. In relation data there is only a data field which will pass through all the data in a serialized way. See here for [provides](https://github.com/canonical/serialized-data-interface/blob/main/provide_interface.py#L25) and here for [requires](https://github.com/canonical/serialized-data-interface/blob/main/require_interface.py#L65d). This way we can keep typing our config options.

**Review by knkski — APPROVED** [review]
:pie:

**Pull request merged**